src/coffeescript.coffee

```diff
@@ -117,7 +117,11 @@ exports.compile = compile = withPrettyErrors (code, options = {}) ->
     sourceCodeNumberOfLines = (code.match(/\r?\n/g) or '').length + 1
     sourceCodeLastLine = /.*$/.exec(code)[0] # `.*` matches all but line break characters.
     ast = nodes.ast options
-    ast.end = ast.range[1] = ast.program.end = ast.program.range[1] = code.length
+    range = [0, code.length]
+    ast.start = ast.program.start = range[0]
```

Comment:
I was seeing that the location data on the root `File`/`Program` nodes was not always starting at `0` when the file started with a comment eg:
```
# comment
"use strict"
```
So similar to what we were already doing to make sure the *end* location data for the root nodes is correct, also make sure their *start* location data is always correct (ie `0`)